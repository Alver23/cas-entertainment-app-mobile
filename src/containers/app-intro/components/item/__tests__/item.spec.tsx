// Dependencies
import React from 'react';
import { Text } from 'react-native-elements';
import { shallow } from 'enzyme';

import Item from '../index';

// Mocks
import mocks from './mocks.json';

describe('Item component', () => {
  let component;

  beforeEach(() => {
    jest.useFakeTimers();
    component = shallow(<Item {...mocks} />);
  });

  it('should rendering correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should rendering correctly title and text', () => {
    const texts = component.find(Text);
    const {
      item: { title, text },
    } = mocks;
    const titleText = texts.findWhere((item) => item.text() === title).exists();

    const descriptionText = texts.findWhere((item) => item.text() === text).exists();

    expect(titleText).toBeTruthy();
    expect(descriptionText).toBeTruthy();
  });
});
