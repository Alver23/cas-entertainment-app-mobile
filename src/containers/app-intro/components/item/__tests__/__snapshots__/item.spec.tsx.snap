// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component should save a snapshot of the component 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 96,
      }
    }
  >
    <Image
      source={
        Object {
          "source": Object {
            "uri": "123",
          },
        }
      }
      style={
        Object {
          "height": 667,
          "marginBottom": 32,
          "resizeMode": "contain",
        }
      }
    />
    <Themed.Text
      h4={true}
    >
      Title
    </Themed.Text>
    <Themed.Text>
      text
    </Themed.Text>
  </View>
</View>
`;
